pkg/3scale/amp/operator/apicast_reconciler.go

```diff
@@ -86,29 +88,31 @@ func (r *ApicastReconciler) Reconcile() (reconcile.Result, error) {
 		reconcilers.DeploymentConfigContainerResourcesMutator,
 		reconcilers.DeploymentConfigAffinityMutator,
 		reconcilers.DeploymentConfigTolerationsMutator,
-		r.apicastProductionWorkersEnvVarMutator,
-		r.apicastLogLevelEnvVarMutator,
-		r.apicastTracingConfigEnvVarsMutator,
-		r.apicastEnvironmentEnvVarMutator,
+		apicastProductionWorkersEnvVarMutator,
```

Comment:
Why has it been changed from APIcastReconciler methods to generic functions?

I understand the decision for the `portsMutator` function, which I guess has been done this way because it could be reusable for other reconcilers, but the other ones seem pretty specific to the APIcastReconciler

Reply:
Just because those functions are global, they do not use anything from the ApicastReconciler struct. All required input is in the input parameters.

Reply:
I understand it from that point of view. I think I would still make them ApicastReconciler methods even in that case because they are specific to it and because they are namespaced to that type instead of being namespaced at the package level, but I am ok with both in any case, not a big deal